File: Java/Algorithms/BubbleSort.java
public class BubbleSort {
  
	public static int[] swap(int[] arr, int x, int j) {
		int temp = arr[x];
		arr[x] = arr[j];
		arr[j] = temp;
		
		return arr;
	}
	public static void main(String[] args) {
		int[] array = {10,2,9,5,3,4,8,1,7};
		boolean swapped = true;
		while(swapped) {
			swapped = false;
			for(int i = 0; i < array.length - 1; i++){
				if(array[i] > array[i + 1]) {
					swap(array, i, i+1);
					swapped= true;
				}
			}
		}
		
		for(int i : array) {
			System.out.println(i);
		}

	}

}

File: Algorithms/Bubble Sort.py

def bubble_sort(arr):
    def swap(i, j):
        arr[i], arr[j] = arr[j], arr[i]
    n = len(arr)
    swapped = True
    x = -1
    while swapped:
        swapped = False
        x = x + 1
        for i in range(1, n-x):
            if arr[i - 1] > arr[i]:
                swap(i - 1, i)
                swapped = True
    return arr

arr = [4,8,3,9,5,8,9,3,4,5,8,2,8,3,9,5,0,2]

print(bubble_sort(arr))

